subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -224,6 +277,211 @@ static int validate_ak_order(void)
 	return 0;
 }
 
+static int bond_name_gen(char *name, uint8_t index)
+{
+	int n;
+
+	n = snprintf(name, SETTINGS_BOND_NAME_MAX_SIZE, "%s%u", SETTINGS_BOND_FULL_PREFIX, index);
+	__ASSERT_NO_MSG(n < SETTINGS_BOND_NAME_MAX_SIZE);
+	if (n < 0) {
+		return n;
+	}
+
+	return 0;
+}
+
+static size_t fp_bond_idx_get(struct fp_bond *bond)
+{
+	return bond - fp_bonds;
+}
+
+static int fp_bond_settings_delete(struct fp_bond *bond)
+{
+	int err;
+	char name[SETTINGS_BOND_NAME_MAX_SIZE];
+
+	err = bond_name_gen(name, fp_bond_idx_get(bond));
+	if (err) {
+		return err;
+	}
+
+	err = settings_delete(name);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int fp_bond_delete(struct fp_bond *bond)
+{
+	int err;
+
+	err = fp_bond_settings_delete(bond);
+	if (err) {
+		return err;
+	}
+
+	*bond = (struct fp_bond){0};
+
+	return 0;
+}
+
+static int bond_remove_completely(struct fp_bond *bond)
+{
+	int err;
+
+	if (bond->bonded) {
+		/* Bond entry will be deleted in the fp_storage_ak_bond_delete function called
+		 * after bond removal from Bluetooth stack.
+		 */
+		err = bt_request_cb->bond_remove(&bond->nv.addr);
+		if (err) {
+			LOG_ERR("Failed to remove Bluetooth bond (err %d)", err);
+			return err;
+		}
+	} else {
+		err = fp_bond_delete(bond);
+		if (err) {
+			LOG_ERR("Failed to delete Fast Pair bond (err %d)", err);
+			return err;
+		}
+	}
+
+	LOG_DBG("Bond removed successfully");
+
+	return 0;
+}
+
+static void bonds_with_invalid_id_remove_completely(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		int err;
+		bool account_key_found = false;
+
+		if (bt_addr_le_eq(&fp_bonds[i].nv.addr, BT_ADDR_LE_ANY) || !fp_bonds[i].bonded ||
+		    (fp_bonds[i].nv.account_key_metadata_id == BOND_AK_ID_UNKNOWN)) {
+			continue;
+		}
+
+		for (size_t j = 0; j < account_key_count; j++) {
+			if (fp_bonds[i].nv.account_key_metadata_id ==
+			    ACCOUNT_KEY_METADATA_FIELD_GET(account_key_metadata[j], ID)) {
+				account_key_found = true;
+				break;
+			}
+		}
+
+		if (!account_key_found) {
+			LOG_DBG("Removing bond associated with invalid Account Key");
+			err = bond_remove_completely(&fp_bonds[i]);
+			if (err) {
+				LOG_ERR("Failed to remove bond (err %d)", err);
+			}
+		}
+	}
+}
+
+static void bonds_conn_ctx_reset(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		fp_bonds[i].conn_ctx = NULL;
+	}
+}
+
+static void bt_bonds_find(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fp_bonds); i++) {
+		fp_bonds[i].bonded = bt_request_cb->is_addr_bonded(&fp_bonds[i].nv.addr);
+		if (fp_bonds[i].bonded) {
+			LOG_DBG("Loaded bond has been found in the Bluetooth bond list");
+		}
```

Comment:
The log is not too informative. Consider either using `LOG_WRN` to indicate case were bond was not found or logging BLE address here (you can also drop the log completely).

Reply:
It helps in debugging.